File: MicroCli/src/main/resources/micronaut/functions/oracle/java/SaveRequestHander.java
package com.example;

import io.micronaut.core.annotation.ReflectiveAccess;
import io.micronaut.oraclecloud.function.OciFunction;
import io.micronaut.oraclecloud.core.TenancyIdProvider;
import jakarta.inject.Inject;
import jakarta.inject.Singleton;


@Singleton
public class Function extends OciFunction {

    @Inject
    TenancyIdProvider tenantIdProvider;

    @ReflectiveAccess
    public String handleRequest() {
        String tenancyId = tenantIdProvider.getTenancyId();
        return "Your tenancy is: " + tenancyId;\
    }
}

File: microstartercli/src/main/resources/micronaut/functions/oracle/kotlin/SaveRequestHander.kt
package com.example

import io.micronaut.core.annotation.ReflectiveAccess
import io.micronaut.oraclecloud.core.TenancyIdProvider
import io.micronaut.oraclecloud.function.OciFunction
import jakarta.inject.Inject
import jakarta.inject.Singleton

@Singleton
class Function : OciFunction() {
    @Inject
    lateinit var tenantIdProvider : TenancyIdProvider

    @ReflectiveAccess
    fun handleRequest() : String {
        val tenancyId = tenantIdProvider.tenancyId
        return "Your tenancy is: $tenancyId"
    }
}
